src/lib/GenericPredictor.cc

```diff
@@ -4,17 +4,23 @@
 
 namespace simeng {
 
-GenericPredictor::GenericPredictor(YAML::Node config)
-    : btbBits_(config["Branch-Predictor"]["BTB-Tag-Bits"].as<uint64_t>()),
-      btb_(1 << btbBits_,
-           {config["Branch-Predictor"]["Fallback-Static-Predictor"]
-                .as<uint16_t>(),
-            0}),
-      satCntBits_(
-          config["Branch-Predictor"]["Saturating-Count-Bits"].as<uint64_t>()),
-      globalHistoryLength_(
-          config["Branch-Predictor"]["Global-History-Length"].as<uint64_t>()),
-      rasSize_(config["Branch-Predictor"]["RAS-entries"].as<uint64_t>()) {
+GenericPredictor::GenericPredictor() {
+  ryml::ConstNodeRef config = config::SimInfo::getConfig();
+  config["Branch-Predictor"]["BTB-Tag-Bits"] >> btbBits_;
+  // Create branch prediction structures
+  config["Branch-Predictor"]["Saturating-Count-Bits"] >> satCntBits_;
+  config["Branch-Predictor"]["Global-History-Length"] >> globalHistoryLength_;
+  config["Branch-Predictor"]["RAS-entries"] >> rasSize_;
+  // Calculate the saturation counter boundary between weakly taken and
+  // not-taken. `(2 ^ num_sat_cnt_bits) / 2` gives the weakly taken state
+  // value
+  uint8_t weaklyTaken = std::pow(2, (satCntBits_ - 1));
```

Comment:
Is there a benefit to using 'std::pow' over '<<' when the power is 2?  I'd have thought '<<' would be faster.

Reply:
From some old issues, we had with a compiler. I've changed it to use `<<` and altered some of the private member variable data widths to better reflect their possible values